src/frontend/middleware/auth.ts

```diff
@@ -1,10 +1,52 @@
+import {UserT, UserWithoutPasswordT} from "~/types/authTypes";
+import {useSnackbarStore} from "~/store/snackbar";
+import {RouteLocationNormalized} from "vue-router";
+import {userStore} from "~/store/user";
 
 export default defineNuxtRouteMiddleware(async (_to, _from) => {
-    const csrftoken = useCookie('csrftoken')
-    const auth = useAuth()
+    // Wait until the user is loaded
+    const user = userStore()
 
-    if (!csrftoken.value || auth.isLoggedIn?.value === false) {
-        csrftoken.value = null
+    // here I need to wait for the user to be fetched or not, which is "user.fetchUserHasBeenRan"
+    if (!user.fetchUserHasBeenRan) {
+
+      try {
+        // Wait for fetch to complete
+        await user.$state.fetchUserHasBeenRan
```

Comment:
is something fancy happening here? Why not just do `user.fetchUseHasBeenRan`? Does the `await` and `$state` actually listen to that property change or something?

Reply:
I think this was just an experiment and doesn't actually do anything?